open-craft/opencraft
# SE-1541 Disable broker heartbeat for celery workers

Celery workers are dodgy and don't appear to be actually sending
heartbeats even when configured (as by default) to a non-zero heartbeat
interval. This instructs them to disable amqp protocol heartbeats
altogether. This, combined with setting heartbeat to 0 on the rabbitmq
server, disables amqp protocol heartbeat checks, and we should avoid
getting any connection reset issues.

This works in conjunction with https://github.com/open-craft/ansible-playbooks/pull/148 and https://github.com/edx/edx-platform/pull/21567 . However, it can be tested without those PRs (we just check that it correctly passes through the variable).

**Test instructions**:

- deploy ocim with this branch
- create a master instance with the default config
- launch an appserver on the instance
- ssh into the appserver and view /edx/etc/lms.yml and /edx/etc/studio.yml and verify that the line `BROKER_HEARTBEAT: 0` is present in both.

**Reviewer**:

- [x] @gr4yscale
